### PR TITLE
Logic Integration: NutritionPlanService and WorkoutLogService

### LINKED ISSUE #97: Logic Integration: NutritionPlanService - Date range calculation
﻿### Task
Integrate logic removed from NutritionPlan.cs.

### Requirements
* Implementation: The service must compute and set StartDate = DateTime.Today and EndDate = DateTime.Today.AddDays(30) explicitly when generating a new plan.

### LINKED ISSUE #98: Logic Integration: WorkoutLogService - Rating handling
﻿### Task
Integrate logic removed from WorkoutLog.cs.

### Requirements
* Implementation: Update the service to handle ratings explicitly. Use double? or set values in the service to avoid the previous -1 sentinel default.

### PR DESCRIPTION
**Pull request by AndaChiorean**
## Summary
- Added NutritionPlanService that explicitly sets StartDate to today and EndDate to 30 days out when creating a new plan, instead of relying on model property defaults
- Added WorkoutLogService with proper nullable rating handling -- ratings of 0 or negative (legacy -1 sentinel) now map to null in the DTO instead of leaking sentinel values through the API
- Changed WorkoutLogDataTransferObject.Rating from double to double? and updated ClientService mapping accordingly
- Set up xUnit test project with Moq and added 9 unit tests covering date defaults, rating mapping edge cases, and DTO round-trips

Closes #97, closes #98

### CODE REVIEW

File: Tests/NutritionPlanServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        Assert.Equal(DateTime.Today.AddDays(30), result.EndDate);
+    }
+
+    [Fact]
+    public async Task CreatePlan_PersistsThroughRepository()
+    {
+        NutritionPlan? capturedPlan = null;
+        this.nutritionRepository
+            .Setup(r => r.InsertNutritionPlanAsync(It.IsAny<NutritionPlan>()))
```
Comment (MarioMaxim18) [inline Tests/NutritionPlanServiceTests.cs]: what is r ?)
read CodeConventions.md

File: UBB-SE-2026-832-1.sln [hunk 2]
```diff
@@ -1,34 +1,76 @@
+﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 17
 VisualStudioVersion = 17.0.31903.59
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ClassLibrary", "ClassLibrary\ClassLibrary.csproj", "{1239CA30-0B9C-4962-B6A2-C8C7A9700B2B}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebAPI", "WebAPI\WebApi.csproj", "{FD9CC266-52C8-4F13-BC6F-C2BC714E6F67}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebApi", "WebAPI\WebApi.csproj", "{FD9CC266-52C8-4F13-BC6F-C2BC714E6F67}"
```
Comment (MarioMaxim18) [inline UBB-SE-2026-832-1.sln]: the directory is correct as WebAPI

**Review by MarioMaxim18 — CHANGES_REQUESTED** [review]

File: Tests/NutritionPlanServiceTests.cs (hunk 1 above)
Comment (AndaChiorean) [reply Tests/NutritionPlanServiceTests.cs]: Good catch, fixed. Lambda parameters now use full names per CodeConventions.

File: UBB-SE-2026-832-1.sln (hunk 2 above)
Comment (AndaChiorean) [reply UBB-SE-2026-832-1.sln]: Fixed -- dotnet sln add silently changed the display name. Restored to WebAPI.

**Review by MarioMaxim18 — APPROVED** [review]
